### PR TITLE
Mark attribute as optional

### LINKED ISSUE #14062: Automation UI editor is not marking "attribute" as optional for "numeric_state"
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

Hello,
just a tiny little bug.
The Automation editor does not show Attribute as an optional field when selecting numeric state as the trigger.

Thank you
Alex

### Describe the behavior you expected

Attribute field reading "Attribute (optional)"

### Steps to reproduce the issue

Open Automation editor in the UI and select numeric state as trigger.

### What version of Home Assistant Core has the issue?

2022.10.3

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

_No response_

### Which operating system are you using to run this browser?

_No response_

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by tradiff**
## Proposed change

Fixes #14062

Indicate that the Attribute field is optional in the automation UI.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

image

- This PR fixes or closes issue: fixes #14062
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Review by piitaya — DISMISSED** [review]

File: src/translations/en.json [hunk 1]
```diff
@@ -399,7 +399,7 @@
           "show_entities": "Show entities"
         },
         "entity-attribute-picker": {
-          "attribute": "Attribute",
+          "attribute": "Attribute (optional)",
```
Comment (frenck) [inline src/translations/en.json:402]: HAving a field optional, should not be part of text or translations of the label of the field.

If that is needed, something is wrong with the UX. As the UI already knows if a field is optional or not.

IMHO this should not be merged, instead the issue should be fixed.

**Review by frenck — CHANGES_REQUESTED** [review]

File: src/translations/en.json (hunk 1 above)
Comment (frenck) [reply src/translations/en.json:402]: 🤷 Sorry didn't know... still think this should not be part of the label translations
The fact the label of the form field is (for example) "from", has nothing to do with it being optional or not.

Comment (travatomic) [reply src/translations/en.json]: @piitaya / @frenck Let me know what you think of my latest commit. Also I haven't added a new translation before, so I'm not sure if I need to also add the key to the other languages. Note: I do not have a lokalise account, so I was not able to run `script/translations_download`. Please advise on whether this is needed to be run by me.

Comment (bramkragten) [reply src/translations/en.json]: According to the material guidelines:
<https://m3.material.io/components/text-fields/guidelines>
CleanShot 2022-10-13 at 11 22 58

Comment (bramkragten) [reply src/translations/en.json]: Since all fields here are optional except for the entity, we should not use the `optional` text here on any field

Comment (piitaya) [reply src/translations/en.json]: Oh the material guidelines are strange... It looks inconsistent to have sometime asterisks, sometime optional labels. 🤔

Comment (piitaya) [reply src/translations/en.json]: @travatomic Just editing translations file is enough 🙂

Comment (travatomic) [reply src/translations/en.json]: I've changed this form to only indicate the required fields, not the optional fields. Updated screenshot:

image

File: src/translations/en.json [hunk 2]
```diff
@@ -1965,7 +1965,7 @@
                   "label": "State",
                   "attribute": "Attribute (optional)",
                   "from": "From (optional)",
-                  "for": "For (optional)",
+                  "for": "For",
```
Comment (travatomic) [inline src/translations/en.json:1968]: The only other place this translation was used is `ha-automation-condition-state.ts`, which is another form that only has `*` for required fields. This was the only field that showed `(optional)`, so this ended up being a good change on that form as well.

**Review by bramkragten — APPROVED** [review]

**Pull request merged**